### ISSUE TITLE
Cursor Lags in Fedora 23

### ISSUE THREAD

**Issue by GizmoCodes** [aseprite/aseprite#1021]
I recently compiled a copy of Aseprite from git (1.1.3-dev) and an old issue (or at least a varient of it) came back. It appears that the cursor seems to lag, or move slowly. It's not a problem with my distro's DE, since it works normally. I don't quite remember how to find a specific git commit version, but according to git I'm up to date with 'origin/master.'

(_Also, when I kinda hi-jacked [another thread](https://github.com/aseprite/aseprite/issues/957), I thought it was the same issue. I didn't mean to offend anyone if I did so, nor did I know that there was a [CoC](https://github.com/aseprite/aseprite/blob/master/CODE_OF_CONDUCT.md). Sorry on that :\ )_

**Comment by nchowning** [comment]
I'm having this same issue on Arch x86_64 with the i3 window manager. It seems like this has crept up before - https://github.com/aseprite/aseprite/issues/283

**Comment by Chae4ek** [comment]
Hello! I think this issue needs more info, and I probably know the reason of these lags.

First, bind any tool to any key (I use 'B' for the Brush, for example), it is important that it's not a 'quick tool'. They seem to work in a different way ('quick tool' also lags, but less).
Then just hold 'B', the mouse will start lagging in a couple of seconds.

It looks to me like the editor handles ***typed*** keys instead of ***pressed*** ones.
The ***typed*** keys are cooked characters and they are typed while you are *holding* the button.
The ***pressed*** keys are key codes, they have their native code for each button as far as I know. They are triggered *only once* when you press the button.
I don't know what kind of terms you use for keystrokes and everything, I hope you understand me.

So, I compiled the lastest beta (v1.3-beta11) on Debian x64 and this problem is still there.
As for mouse moving without holding the keys, I didn't notice any lags. I don't know if it could be that the moving is read in a similar way.
Sorry if it doesn't apply to this issue, should I open a new issue then?

**Comment by dacap** [comment]
Hi @Chae4ek, it looks like your lag problem is a different one. Generally we interpret key presses/releases, and "typed characters" are interpreted only on text input fields. At least that should work in that way, you can test this pressing the B key to select the Pencil tool (keep the B key pressed) and then press the E key (without releasing the B key). The eraser tool should be selected, but the Pencil tool will be selected only after you press the B key again (at least that is how it works on Windows and macOS).
